Makefile

```diff
@@ -1,55 +1,46 @@
 GO ?= go
 CC ?= gcc
-ifeq ($(GOPATH),)
-export GOPATH := $(shell $(GO) env GOPATH)
-endif
-FIRST_GOPATH := $(firstword $(subst :, ,$(GOPATH)))
-GOBIN := $(shell $(GO) env GOBIN)
-ifeq ($(GOBIN),)
-	GOBIN := $(FIRST_GOPATH)/bin
-endif
 
-all: build test phaul phaul-test
+all: build test phaul-test
 
 lint:
-	@golint -set_exit_status . test phaul
+	golangci-lint run ./...
```

Comment:
What about enabling some of the linters presets?

```
golangci-lint run -p bugs,performance,unused,format ./...
```

Reply:
I think this should be addressed separately (and it's better/easier to do so via `.golangci.yml`), once this is merged.

Reply:
@rst0git implemented; PTAL

Reply:
This probably makes the patchset slightly harder to review -- let me know if you want a separate PR.

Reply:
Thanks!